devel/capnproto/Portfile

```diff
@@ -47,6 +47,25 @@ configure.cppflags-replace \
 if {[string match *clang* ${configure.compiler}]} {
     configure.cxx_stdlib        libc++
     configure.ldflags-append    -stdlib=libc++
+} elseif {[string match *gcc* ${configure.compiler}]} {
+    # Partly merged: https://github.com/capnproto/capnproto/pull/1711
+    # Re other hacks, see:
+    # https://github.com/capnproto/capnproto/issues/1710
+    # https://github.com/capnproto/capnproto/issues/1712
+    # With these, 946 test cases pass, 11 fail.
+    # FIXME: judging by test results, mutices and filesystem require attention.
+    patchfiles-append   0001-message.h-fix-for-Darwin-ppc.patch \
+                        0002-filesystem.c-a-hack-to-fix-overloaded-set.patch \
+                        0003-async-io-unix.c-fix-for-missing-SOL_LOCAL.patch \
+                        0004-capability.c-a-hack-for-capability.patch \
+                        0005-Darwin-PPC-disable-negative-broken-asserts.patch \
+                        0006-capability-test-fix-for-macOS-with-GCC.patch \
+                        0007-mutex.c-fix-for-Apple.patch
```

Comment:
Since those only apply on systems that compile with gcc, that seems OK with me.

Reply:
Hope with the help of upstream we can fix those in a better way and possibly mergeable, though perhaps not for filesystem-disk-unix, since it has to rely on legacysupport for several functions used there.